geode-core/src/distributedTest/java/org/apache/geode/cache/ConnectionPoolDUnitTest.java

```diff
@@ -4219,360 +2990,254 @@ public void run2() throws CacheException {
-          ctl.waitForUpdated(k3);
-          assertEquals(v1, r.getEntry(k3).getValue()); // Ensure that the notifier updated the entry
-        }
-      });
-      // Ok, now we are ready to do some dynamic region action!
-      final String v1Dynamic = v1 + "dynamic";
-      final String dynFromClientName = name + "-dynamic-client";
-      final String dynFromServerName = name + "-dynamic-server";
-      client1.invoke("Client dynamic region creation", new CacheSerializableRunnable() {
-        @Override
-        public void run2() throws CacheException {
-          assertTrue(DynamicRegionFactory.get().isOpen());
-          Region<Object, Object> r = getRootRegion(name);
-          assertNotNull(r);
-          Region<Object, Object> dr =
-              DynamicRegionFactory.get().createDynamicRegion(name, dynFromClientName);
-          assertNull(dr.get(k1)); // This should be enough to validate the creation on the server
-          dr.put(k1, v1Dynamic);
-          assertEquals(v1Dynamic, dr.getEntry(k1).getValue());
-        }
-      });
+    srv1.invoke("Validate dynamic region destruction on other server", () -> {
+      Region<Object, Object> r = getRootRegion(name);
+      assertThat(r).isNotNull();
+      String drName = r.getFullPath() + Region.SEPARATOR + dynFromServerName;
 
-      // Assert the servers have the dynamic region and the new value
-      CacheSerializableRunnable valDR =
-          new CacheSerializableRunnable("Validate dynamic region creation on server") {
-            @Override
-            public void run2() throws CacheException {
-              Region<Object, Object> r = getRootRegion(name);
-              assertNotNull(r);
-              long end = System.currentTimeMillis() + 10000;
-              Region dr;
-              for (;;) {
-                try {
-                  dr = r.getSubregion(dynFromClientName);
-                  assertNotNull(dr);
-                  assertNotNull(getCache().getRegion(name + Region.SEPARATOR + dynFromClientName));
-                  break;
-                } catch (AssertionError e) {
-                  if (System.currentTimeMillis() > end) {
-                    throw e;
-                  }
-                }
-              }
-
-              assertEquals(v1Dynamic, dr.getEntry(k1).getValue());
-            }
-          };
-      srv1.invoke(valDR);
-      srv2.invoke(valDR);
-      // now delete the dynamic region and see if it goes away on servers
-      client1.invoke("Client dynamic region destruction", new CacheSerializableRunnable() {
-        @Override
-        public void run2() throws CacheException {
-          assertTrue(DynamicRegionFactory.get().isActive());
-          Region<Object, Object> r = getRootRegion(name);
-          assertNotNull(r);
-          String drName = r.getFullPath() + Region.SEPARATOR + dynFromClientName;
+      await().pollInterval(100, MILLISECONDS).until(() -> getCache().getRegion(drName) == null);
```

Comment:
I recommend not using `pollInterval` just to keep these statements simpler and uniform.